imqsauth/http.go

```diff
@@ -770,3 +847,72 @@ func httpHandlerGetGroups(central *ImqsCentral, w http.ResponseWriter, r *httpRe
 		httpSendGroupsJson(w, groups)
 	}
 }
+
+func (central *ImqsCentral) handleWebSocket(ws *websocket.Conn) {
+	closeConnection := true
+	if token, err := authaus.HttpHandlerPrelude(&central.Config.Authaus.HTTP, central.Central, ws.Request()); err == nil {
+		if permList, errDecodePerms := authaus.PermitResolveToList(token.Permit.Roles, central.Central.GetRoleGroupDB()); errDecodePerms == nil {
+			closeConnection = !permList.Has(PermEnabled)
+		}
+	}
+
+	if closeConnection {
+		ws.Close()
+		return
+	}
+
+	id := central.addSubscriber(ws)
+
+	central.Central.Log.Infof("Added auth web socket client %d", id)
+	var msg string
+	for {
+		// We want to timeously close and remove the web sockets that send
+		// EOF's or invalid data, not wait until we fail to send and only then close them
+		// We just ignore whatever else we receive
+		if err := websocket.Message.Receive(ws, &msg); err != nil {
+			closeAndDeleteWebSocket(central, id)
+			return
+		}
+	}
+}
+
+func closeAndDeleteWebSocket(central *ImqsCentral, id uint64) {
+	central.Lock()
+	defer central.Unlock()
+
+	ws, ok := central.RoleChangeSubscribers[id]
+	if ok {
+		ws.Close()
+	}
+	delete(central.RoleChangeSubscribers, id)
+}
+
+func (central *ImqsCentral) broadcastToAllSubscribers(msg string) {
+	var deleteIDs []uint64
+	central.RLock()
+
+	for id, ws := range central.RoleChangeSubscribers {
+		err := websocket.Message.Send(ws, msg)
+		if err != nil {
+			// only record the id, since we already have a read lock active at this point
+			deleteIDs = append(deleteIDs, id)
+		}
+	}
+
+	central.RUnlock()
+
+	// now delete the recorded id's
+	for _, id2 := range deleteIDs {
```

Comment:
No need to call this `id2`. Rather just use `id` again. Variables introduced inside a 'for' statement have the same scope as the loop.